### PR TITLE
Cleaning up KBO

### PR DESCRIPTION
**Pull request by ibnyusuf**
PR achieves the following:

- Setting different symbol precedence schemes affects type constructors as well (previously they always used default scheme)
- There can now only be a single unary function symbol of weight zero (in line with the theory). Exceptions are made for unary minus symbols until new calculus can be merged.
- For the purpose of the term ordering, a function symbol is considered constant or unary based on the number of _term_ arguments it accepts (in line with the theory).

### CODE REVIEW

File: Kernel/KBO.cpp [hunk 1]
```diff
@@ -567,72 +554,71 @@ KBO KBO::testKBO()
...
+  for (FunctionSymbol i = 1; i < nFunctions; i++) {
     if (compareFunctionPrecedences(maxFn, i) == LESS) {
-      maximalFunctions.replace(sort, i);
+      maxFn = i;
     }
   }
 
+  // TODO remove unary minus check once new 
+  // theory calculus goes into master
+  auto isUnaryMinus = [](unsigned functor){
```
Comment (joe-hauns) [inline Kernel/KBO.cpp:606]: I locally made some changes so that the unary minus doesn't get added to the signature here, if it is not present in the problem. I'm not sure whether i should push it to this branch or make a separate PR. What do you think?
It's a quite small thing.

**Review by joe-hauns — APPROVED** [review]
Looks all good to me. 
I added some helpers to get rid of some copy-paste code i wrote at some point.

File: Kernel/Ordering.cpp [hunk 2]
```diff
@@ -891,40 +901,42 @@ DArray<int> PrecedenceOrdering::predLevelsFromOptsAndPrec(Problem& prb, const Op
...
-    out << "% ===== end of function precedences ===== " << std::endl;
-  }
+      out << "%" << std::endl;
+    };
 
-  out << "%" << std::endl;
+  _show("type constructor", 
+      env.signature->typeCons(), 
+      [](unsigned f) { return env.signature->getTypeCon(f); },
+      [&](unsigned l, unsigned r){ return intoComparison(compareFunctionPrecedences(l,r)); });
```
Comment (joe-hauns) [inline Kernel/Ordering.cpp:927]: ah yeah .. is this the correct way of comparing type constructor precedences?
I copied it from your code, but I reckon that should be compareTypeConPrecedences, right?

File: Kernel/KBO.cpp (hunk 1 above)
Comment (ibnyusuf) [reply Kernel/KBO.cpp:606]: Don't have a preference either way. Adding it to this PR is fine by me.

File: Kernel/Ordering.cpp (hunk 2 above)
Comment (ibnyusuf) [reply Kernel/Ordering.cpp]: Thanks, fixed!

File: Kernel/KBO.cpp (hunk 1 above)
Comment (quickbeam123) [reply Kernel/KBO.cpp:606]: Yes, @joe-hauns, please add here if you haven't already. Thank you!
